src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,665 @@
+import matplotlib.pyplot as plt
+
+from .base import WorkflowBase
+from ..parsing import gmx, amber
+from ..preprocessing.subsampling import decorrelate_dhdl, decorrelate_u_nk
+from ..estimators import BAR, TI
+from ..estimators import AutoMBAR as MBAR
+from ..visualisation import (plot_mbar_overlap_matrix, plot_ti_dhdl,
+                             plot_dF_state, plot_convergence)
+from ..postprocessors.units import get_unit_converter
+from ..convergence import forward_backward_convergence
+from .. import concat
+from .. import __version__
+
+
+class ABFE(WorkflowBase):
+    '''Alchemical Analysis style automatic workflow.
+
+    Parameters
+    ----------
+    units : str
+        The unit used for printing and plotting results. {'kcal/mol', 'kJ/mol',
+        'kT'}. Default: 'kT'.
+    software : str
+        The software used for generating input. {'GROMACS', 'AMBER'}
+    dir : str
+        Directory in which data files are stored. Default: os.path.curdir.
+    prefix : str
+        Prefix for datafile sets. Default: 'dhdl'.
+    suffix : str
+        Suffix for datafile sets. Default: 'xvg'.
+    T : float
+        Temperature in K. Default: 298.
+    outdirectory : str
+        Directory in which the output files produced by this script will be
+        stored. Default: os.path.curdir.
+
+    Attributes
+    ----------
+    logger : Logger
+        The logging object.
+    file_list : list
+        The list of filenames sorted by the lambda state.
+    '''
+    def __init__(self, units='kT', software='Gromacs', dir=os.path.curdir,
+                 prefix='dhdl', suffix='xvg', T=298, outdirectory=os.path.curdir):
+
+        super().__init__(units, software, T, outdirectory)
+        self.logger = logging.getLogger('alchemlyb.workflows.ABFE')
+        self.logger.info('Initialise Alchemlyb ABFE Workflow')
+        self.logger.info(f'Alchemlyb Version: f{__version__}')
+        self.logger.info(f'Set Temperature to {T} K.')
+        self.logger.info(f'Set Software to {software}.')
+
+        self.update_units(units)
+
+        self.logger.info(f'Finding files with prefix: {prefix}, suffix: '
+                         f'{suffix} under directory {dir} produced by '
+                         f'{software}')
+        self.file_list = glob(dir + '/**/' + prefix + '*' + suffix,
+                         recursive=True)
+
+        self.logger.info(f'Found {len(self.file_list)} xvg files.')
+        self.logger.info("Unsorted file list: \n %s", '\n'.join(
+            self.file_list))
+
+        if software.lower() == 'gromacs':
+            self.logger.info(f'Using {software} parser to read the data.')
+            self._extract_u_nk = gmx.extract_u_nk
+            self._extract_dHdl = gmx.extract_dHdl
+        elif software.lower() == 'amber':
+            self._extract_u_nk = amber.extract_u_nk
+            self._extract_dHdl = amber.extract_dHdl
+        else: # pragma: no cover
+            raise NameError(f'{software} parser not found.')
```

Comment:
This is the wrong type of exception. It's a `ValueError` (wrong value).

Instead of "not found" say "not implemented".